### ISSUE TITLE
[PostgreSQL] default_sequence_name raises an error when the table has composite primary key

### ISSUE THREAD

**Issue by h6ah4i** [rails/rails#53048]
Hi, I noticed that the `default_sequence_name` method raises an error when the table has composite primary key.

### ActiveRecord version:
7.1.4

### Error message:
```
can't quote Array
```

### The `default_sequence_name` method implementation:
https://github.com/rails/rails/blob/6f57590388ca38ed2b83bc1207a8be13a9ba2aef/activerecord/lib/active_record/connection_adapters/postgresql/schema_statements.rb#L249-L259

**Comment by fatkodima** [comment]
Can you please provide a reproduction test script using https://github.com/rails/rails/blob/main/guides/bug_report_templates/active_record.rb template?

**Comment by h6ah4i** [comment]
@fatkodima Hi. Here's a reproduction test for this issue.

### Setup database (PostgreSQL)

```sh
docker run --env POSTGRES_PASSWORD=password --env POSTGRES_DB=issue_53048_repro -p 5432:5432 postgres:latest
```

### Test code

```ruby
# frozen_string_literal: true

require "bundler/inline"

gemfile(true) do
  source "https://rubygems.org"

  gem "rails"
  gem "pg"
  gem "net-protocol", "~> 0.2.2"
end

require "active_record"
require "minitest/autorun"
require "logger"

ActiveRecord::Base.establish_connection(adapter: "postgresql", database: "issue_53048_repro", host: "localhost", port: 5432, user: "postgres", password: "password")
ActiveRecord::Base.logger = Logger.new(STDOUT)

ActiveRecord::Schema.define do
  create_table :products, primary_key: [:store_id, :sku], force: true do |t|
    t.integer :store_id
    t.string :sku
  end
end

class Product < ActiveRecord::Base
end

class BugTest < ActiveSupport::TestCase
  def test_issue_53048
    Product.sequence_name
  end
end
```

**Comment by fatkodima** [comment]
`default_sequence_name` is a private method (declared using `# :nodoc:`). In either case, it will return `nil`, because there is no sequence for this table. Why do you need to use it?

**Comment by h6ah4i** [comment]
> default_sequence_name is a private method (declared using # :nodoc:).



I actually encountered this issue while using the    
[activerecord-import](https://github.com/zdennis/activerecord-import) gem. This gem internally calls the model class's `sequence_name` method, which triggers an error within the `default_sequence_name` method.

> In either case, it will return nil, because there is no sequence for this table.


The reproduction code I posted doesn't involve any sequences, but I can assure you that it still reproduces the same error.
